### PR TITLE
[CLN] website_sale: js cleanup

### PR DESCRIPTION
**Pull request by Feyensv**
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/static/src/js/cart.js [hunk 1]
```diff
@@ -1,17 +1,53 @@
...
+
+import wSaleUtils from "@website_sale/js/website_sale_utils";
+
 
 publicWidget.registry.websiteSaleCart = publicWidget.Widget.extend({
-    selector: '.oe_website_sale .oe_cart',
+    selector: '#shop_cart',
     events: {
+        'change input.js_quantity[data-product-id]': '_onChangeCartQuantity',
         'click .js_delete_product': '_onClickDeleteProduct',
```
[the remark this answers is not in the document]
Comment (vchu-odoo) [reply addons/website_sale/static/src/js/cart.js:15]: Indeed, thank you for notifying me!

File: addons/website_sale/static/src/js/cart.js [hunk 2]
```diff
@@ -1,17 +1,53 @@
 /** @odoo-module **/
 
 import publicWidget from "@web/legacy/js/public/public_widget";
+import { rpc } from "@web/core/network/rpc";
```
Comment (vchu-odoo) [inline addons/website_sale/static/src/js/cart.js:3]: should not we use single quotes in js files as well?

Comment (Feyensv) [reply addons/website_sale/static/src/js/cart.js:3]: I've personally decided to stop fighting against js team on that topic, and to follow the current standard of the file I'm working in.

File: addons/website_sale/static/src/js/website_sale.js [hunk 3]
```diff
@@ -1,33 +1,28 @@
...
+import { extraMenuUpdateCallbacks } from "@website/js/content/menu";
 import { ProductImageViewer } from "@website_sale/js/components/website_sale_image_viewer";
-import { rpc } from "@web/core/network/rpc";
-import { debounce, throttleForAnimation } from "@web/core/utils/timing";
-import { listenSizeChange, SIZES, utils as uiUtils } from "@web/core/ui/ui_service";
-import { isBrowserFirefox, hasTouch } from "@web/core/browser/feature_detection";
-import { Component } from "@odoo/owl";
+import VariantMixin from "@website_sale/js/sale_variant_mixin";
+import { cartHandlerMixin } from "@website_sale/js/website_sale_utils";
+
```
Comment (chevalierv) [inline addons/website_sale/static/src/js/website_sale.js:14]: nitpicking
```suggestion
import { hasTouch, isBrowserFirefox } from "@web/core/browser/feature_detection";
import { rpc } from "@web/core/network/rpc";
import { SIZES, utils as uiUtils } from "@web/core/ui/ui_service";
import { throttleForAnimation } from "@web/core/utils/timing";
import publicWidget from "@web/legacy/js/public/public_widget";
import { extraMenuUpdateCallbacks } from "@website/js/content/menu";
import "@website/libs/zoomodoo/zoomodoo";
import { ProductImageViewer } from "@website_sale/js/components/website_sale_image_viewer";
import VariantMixin from "@website_sale/js/sale_variant_mixin";
import { cartHandlerMixin } from "@website_sale/js/website_sale_utils";
```

File: addons/website_sale/static/src/js/website_sale.js [hunk 4]
```diff
@@ -1,33 +1,28 @@
 /** @odoo-module **/
```
Comment (chevalierv) [inline addons/website_sale/static/src/js/website_sale.js:1]: We could take this opportunity to remove this now-useless header :)
```suggestion
```

**Review by chevalierv — APPROVED** [review]
Hello @Feyensv 

Thanks for clearing eCommerce! LGTM 👍 

Have a nice day!

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

File: addons/website_sale/static/src/js/cart.js [hunk 5]
```diff
@@ -1,17 +1,51 @@
...
     events: {
+        'change input.js_quantity[data-product-id]': '_onChangeCartQuantity',
         'click .js_delete_product': '_onClickDeleteProduct',
+        'click a.js_add_suggested_products': '_onClickSuggestedProduct',
+    },
+
+    /**
+     * @constructor
+     */
+    init: function () {
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/cart.js:20]:
```suggestion
    init() {
```

File: addons/website_sale/static/src/js/cart.js [hunk 6]
```diff
@@ -1,17 +1,51 @@
...
 
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onChangeCartQuantity: function (ev) {
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/cart.js:34]:
```suggestion
    _onChangeCartQuantity(ev) {
```

File: addons/website_sale/static/src/js/cart.js [hunk 7]
```diff
@@ -1,17 +1,51 @@
...
+        var $input = $(ev.currentTarget);
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/cart.js:35]: `var` -> `let`|`const`

Comment (AntoineVDV) [inline addons/website_sale/static/src/js/cart.js:35]: Drop jQuery.

File: addons/website_sale/static/src/js/cart.js (hunk 6 above)
Comment (AntoineVDV) [reply addons/website_sale/static/src/js/cart.js:34]: Same for other functions

File: addons/website_sale/static/src/js/cart.js [hunk 8]
```diff
@@ -20,6 +54,50 @@ publicWidget.registry.websiteSaleCart = publicWidget.Widget.extend({
...
+     * @private
+     */
+    _changeCartQuantity: function ($input, value, $dom_optional, line_id, productIDs) {
+        $($dom_optional).toArray().forEach((elem) => {
+            $(elem).find('.js_quantity').text(value);
+            productIDs.push($(elem).find('span[data-product-id]').data('product-id'));
+        });
+        $input.data('update_change', true);
+
+        rpc('/shop/cart/update_json', {
```
Comment (AntoineVDV) [inline addons/website_sale/static/src/js/cart.js:74]: `data = await rpc(...)`

**Pull request closed without merging**